azure_pipelines/spm-framework.yml

```diff
@@ -0,0 +1,311 @@
+
+        chlg="## Release Notes"$'\n'
+        if [ ! -e changelog.txt ]; then
+          echo "CHANGELOG NOT FOUND!"
+        else
+          fl=0
+          while read p; do
+            if [[ "$p" =~ 'Version' ]]; then
+              fl=$((fl + 1))
+              if [ $fl -gt 1 ]
+              then
+                  break
+              fi
+            fi
+
+            if [[ ($fl -eq 1)  && !("$p" =~ ^\*\s+) ]]; then
+              chlg="$chlg"$'\n'"$p"
+            fi
+          done < "MSAL/IdentityCore/changelog.txt"
+        fi
+        echo "chlg = ${chlg}"
+        echo "${chlg}" > cc-release-notes.md
+
+      workingDirectory: '$(Build.SourcesDirectory)'
+      failOnStderr: true
+      noProfile: false
+      noRc: false
+  - task: GitHubRelease@1
+    displayName: Generate Common Core Github Release
+    inputs:
+      gitHubConnection: '$(GithubServiceConnection)'
+      repositoryName: 'AzureAD/microsoft-authentication-library-common-for-objc'
+      action: 'create'
+      target: 'master'
+      tagSource: 'userSpecifiedTag'
+      tag: '$(commonCoreReleaseVersion)'
+      title: '$(commonCoreReleaseVersion)'
+      releaseNotesFilePath: 'cc-release-notes.md'
+      changeLogCompareToRelease: 'lastFullRelease'
+      changeLogType: 'issueBased'
+      isDraft: true
+  - task: GitHubRelease@1
+    displayName: Generate MSAL Github Release
+    inputs:
+      gitHubConnection: '$(GithubServiceConnection)'
+      repositoryName: '$(repositoryName)'
+      action: 'create'
+      target: '$(repositoryBranch)'
+      tagSource: 'userSpecifiedTag'
+      tag: '$(releaseVersion)'
+      title: '$(releaseVersion)'
+      releaseNotesFilePath: 'release-notes.md'
+      assets: |
+        $(Build.ArtifactStagingDirectory)/*.zip
+        $(Build.ArtifactStagingDirectory)/*.tar.gz
+      changeLogCompareToRelease: 'lastFullRelease'
+      changeLogType: 'issueBased'
+      isDraft: true
+  - task: Bash@3
+    displayName: Update package.swift with url & checksum & git push 
+    inputs:
+      targetType: 'inline'
+      script: |
+        rm -rf release-notes.md
+        rm -rf cc-release-notes.md
+        git fetch origin $(repositoryBranch) -q
+        git checkout FETCH_HEAD -q
+        git checkout -b update-package-for-$(releaseVersion) -q
+        if [ ! -e Package.swift ]; then
+            echo -e "// swift-tools-version:5.3\n" >> Package.swift
+            cat >> Package.swift << EOF
+        import PackageDescription
+
+        let package = Package(
+          name: "MSAL",
+          platforms: [
+                .macOS(.v10_12),.iOS(.v11)
+          ],
+          products: [
+              .library(
+                  name: "MSAL",
+                  targets: ["MSAL"]),
+          ],
+          targets: [
+              .binaryTarget(name: "MSAL", url: "https://github.com/$(repositoryName)/releases/download/releaseTag1.2.3/MSAL.zip", checksum: "abcdefabcdef1234567890abcdef1234567890abcdef1234567890abcdef1234")
```

Comment:
Does checksum need to be dynamic here?

Reply:
This is just a garbage checksum value I've added initially. This task will add Package.swift to the repo if it does not exist with this garbage value and in the next step the value will be replaced with the actual checksum of the signed & zipped framework.